treeherder/etl/buildapi.py

```diff
@@ -307,6 +334,13 @@ def transform(self, data, source, filter_to_revision=None, filter_to_project=Non
                 # using project and revision form the revision lookups
                 # to filter those jobs with unmatched revision
                 for job in jobs:
+                    job_ids_seen_now.add(job['id'])
```

Comment:
Yeah good suggestion :-)
I think I'll make this change, the only thing is that post bug 1162947 it may have to be undone, for the same reason the the one below.